File: spark/example1/SimpleApp.py
"""SimpleApp.py"""
from pyspark.sql import SparkSession

logFile = "/Users/fenghaojiang/spark/spark-3.2.1/README.md"  # Should be some file on your system
spark = SparkSession.builder.appName("simpleApp").master("local").getOrCreate()
logData = spark.read.text(logFile).cache()

numAs = logData.filter(logData.value.contains('a')).count()
numBs = logData.filter(logData.value.contains('b')).count()

print("Lines with a: %i, lines with b: %i" % (numAs, numBs))

spark.stop()

File: spark/example1/src/main/scala/SimpleApp.scala
import org.apache.spark.sql.SparkSession

object SimpleApp {
    def main(args: Array[String]) = {
        val logFile = "/Users/fenghaojiang/spark/spark-3.2.1/README.md"
        val spark = SparkSession.builder.appName("Simple Application").getOrCreate()
        val logData = spark.read.textFile(logFile)
        val numAs = logData.filter(line => line.contains("a")).count()
        val numBs = logData.filter(line => line.contains("b")).count()
        println(s"Lines with a: $numAs, lines with b: $numBs")
        spark.stop()
    }
}